packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -0,0 +1,168 @@
+import { taskViewService } from '../statuses';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath, hasRootWorkspace, OrgAuthInfo } from '../util';
+
+export const folderTypes = new Set(['EmailTemplate', 'Reports']);
+export class ForceListMetadataExecutor extends SfdxCommandletExecutor<string> {
+  private metadataType: string;
+  private outputPath: string;
+  private defaultUsernameOrAlias: string;
+
+  public constructor(
+    metadataType: string,
+    outputPath: string,
+    defaultUsernameOrAlias: string
+  ) {
+    super();
+    this.metadataType = metadataType;
+    this.outputPath = outputPath;
+    this.defaultUsernameOrAlias = defaultUsernameOrAlias;
+  }
+
+  public build(data: {}): Command {
+    let builder = new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_list_metadata_text'))
+      .withArg('force:mdapi:listmetadata')
+      .withFlag('-m', this.metadataType)
+      .withFlag('-u', this.defaultUsernameOrAlias)
+      .withFlag('-f', this.outputPath)
+      .withJson();
+
+    if (folderTypes.has(this.metadataType)) {
+      builder = builder.withFlag('--folder', 'unfiled$public');
+    }
+    return builder.build();
+  }
+
+  public execute(response: ContinueResponse<string>): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async data => {
+      this.logMetric(execution.command.logName, startTime);
+      buildComponentsList(this.outputPath, this.metadataType);
+    });
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+const workspaceChecker = new SfdxWorkspaceChecker();
+const parameterGatherer = new EmptyParametersGatherer();
+
+export async function forceListMetadata(
+  metadataType: string,
+  defaultUsernameOrAlias: string
+) {
+  const outputPath = await getComponentsPath(
+    metadataType,
+    defaultUsernameOrAlias
+  );
+  const describeExecutor = new ForceListMetadataExecutor(
+    metadataType,
+    outputPath,
+    defaultUsernameOrAlias
+  );
+  const commandlet = new SfdxCommandlet(
+    workspaceChecker,
+    parameterGatherer,
+    describeExecutor
+  );
+  await commandlet.run();
+}
+
+export async function getComponentsPath(
+  metadataType: string,
+  defaultUsernameOrAlias: string
+): Promise<string> {
+  if (hasRootWorkspace()) {
+    try {
+      const workspaceRootPath = getRootWorkspacePath();
+      const username = await OrgAuthInfo.getUsername(defaultUsernameOrAlias);
+
+      if (isNullOrUndefined(username)) {
```

Comment:
Added this since now `OrgAuthInfo.getUsername(...)` returns string or undefined.